lib/jsdom/living/xmlhttprequest.js

```diff
@@ -585,6 +585,8 @@ module.exports = function createXMLHttpRequest(window) {
             }
           } else if (body instanceof ArrayBuffer) {
             flag.body = new Buffer(new Uint8Array(body));
+          } else if (ArrayBuffer.isView(body)) {
+            flag.body = new Buffer(body.buffer);
```

Comment:
You need `new Buffer(body.buffer, body.byteOffset, body.byteLength)` to make it work with non-default views on a ArrayBuffer, like pooled Node.js `Buffer` objects.